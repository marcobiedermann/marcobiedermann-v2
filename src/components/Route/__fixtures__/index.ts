const route = {
  id: 'b6c3ceb6-d89f-4a03-9e96-77768f278909',
  name: 'Page name',
  path: '/path/to/page',
};

export default route;
