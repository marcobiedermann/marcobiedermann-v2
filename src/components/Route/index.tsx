import { Link } from 'gatsby';
import React from 'react';

export interface RouteProps {
  id: string;
  className?: string;
  name: string;
  path: string;
}

const Route: React.FC<RouteProps> = props => {
  const { className, name, path } = props;

  return (
    <Link className={className} to={path}>
      {name}
    </Link>
  );
};

export default Route;
