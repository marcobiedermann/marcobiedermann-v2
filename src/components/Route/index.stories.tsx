/* eslint-disable import/no-extraneous-dependencies */

import { storiesOf } from '@storybook/react';
import React from 'react';
import Route from '.';
import route from './__fixtures__';

storiesOf('Route', module).add('default', () => <Route {...route} />);
