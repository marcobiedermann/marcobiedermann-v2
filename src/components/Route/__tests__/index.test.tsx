/* eslint-disable import/no-extraneous-dependencies */

import { render } from '@testing-library/react';
import React from 'react';
import Route from '..';
import route from '../__fixtures__';

describe('Route component', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<Route {...route} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
