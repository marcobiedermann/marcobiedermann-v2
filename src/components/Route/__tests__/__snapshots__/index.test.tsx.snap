// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Route component should render correctly 1`] = `
<DocumentFragment>
  <a
    href="/path/to/page"
    id="b6c3ceb6-d89f-4a03-9e96-77768f278909"
  >
    Page name
  </a>
</DocumentFragment>
`;
